subt/ros/robot/src/laserscan_to_pointcloud.py

```diff
@@ -0,0 +1,23 @@
+#!/usr/bin/python
+
+import rospy
+from sensor_msgs.msg import PointCloud2, LaserScan
+from laser_geometry import LaserProjection
+
+class LaserScanToPointCloud:
+    def __init__(self):
+        self.laserProj = LaserProjection()
+        self.pointCloudPublisher = rospy.Publisher("/points", PointCloud2, queue_size = 1)
+        self.laserScanSubscriber = rospy.Subscriber("/scan", LaserScan, self.laserScanCallback)
+
+    def laserScanCallback(self, data):
+        cloudOut = self.laserProj.projectLaser(data)
```

Comment:
For my education: Does this also fill in the header, such as time stamp and reference frame?

Reply:
Yes. It uses values from the header of the original LaserScan